pkg/crc/segment/segment.go

```diff
@@ -0,0 +1,82 @@
+package segment
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/pborman/uuid"
+	"github.com/segmentio/analytics-go"
+)
+
+type Client struct {
+	segmentClient     analytics.Client
+	config            *crcConfig.Config
+	telemetryFilePath string
+}
+
+func NewClient(config *crcConfig.Config, telemetryFilePath string) (*Client, error) {
+	client, err := analytics.NewWithConfig("cvpHsNcmGCJqVzf6YxrSnVlwFSAZaYtp", analytics.Config{
+		DefaultContext: &analytics.Context{
+			App: analytics.AppInfo{
+				Name:    constants.DefaultName,
```

Comment:
Imo `constants.DefaultName` is one of these constants that we should not be using, but try to remove.

Reply:
@cfergeau and what do you propose, passing it's value to the NewClient side, make it more explicit ?

Reply:
Passing it to NewClient would probably make sense yes. Maybe it can be improved in a follow-up PR.